rxandroid/src/main/java/rx/android/schedulers/HandlerScheduler.java

```diff
@@ -44,45 +45,49 @@ public Worker createWorker() {
     }
 
     static class HandlerWorker extends Worker {
-
-        final Handler handler;
-
-        private final CompositeSubscription compositeSubscription = new CompositeSubscription();
+        private final Handler handler;
+        /* Writes are guarded by 'this'. */
+        private volatile boolean unsubscribed;
 
         HandlerWorker(Handler handler) {
```

Comment:
oh nice, now I see 👍 

// stuck to idea that `Handler` per `Worker` is required, your approach is better